src/eva/vision/data/datasets/segmentation/consep.py

```diff
@@ -0,0 +1,139 @@
+"""CoNSeP segmentation dataset class."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Literal, Tuple
+
+import torch
+from torchvision import tv_tensors
+from typing_extensions import override
+
+from eva.vision.data.datasets import _validators
+from eva.vision.data.datasets.segmentation import base
+from eva.vision.utils import io
+
+
+class CoNSeP(base.ImageSegmentation):
+    """CoNSeP segmentation dataset."""
+
+    _expected_dataset_lengths: Dict[str | None, int] = {
+        "train": 27,
+        "val": 14,
+        None: 41,
+    }
+    """Dataset split to the expected size."""
+
+    def __init__(
+        self,
+        root: str,
+        split: Literal["train", "val"] | None,
+        transforms: Callable | None = None,
+    ) -> None:
+        """Initialize dataset.
+
+        Args:
+            root: Path to the root directory of the dataset. The dataset will
+                be downloaded and extracted here, if it does not already exist.
+            split: Dataset split to use. If `None`, the entire dataset is used.
+            classes: Whether to configure the dataset with a subset of classes.
+                If `None`, it will use all of them.
+            transforms: A function/transforms that takes in an image and a target
+                mask and returns the transformed versions of both.
+        """
+        super().__init__(transforms=transforms)
+
+        self._root = root
+        self._split = split
+        self._samples_names: List[str] = []
+        self._indices: List[Tuple[int, int]] = []
+
+    @property
+    @override
+    def classes(self) -> List[str]:
+        return [
+            "other",
+            "inflammatory",
+            "healthyepithelial",
+            "dysplastic/malignant epithelial",
+            "fibroblast",
+            "muscle",
+            "endothelial",
+        ]
+
+    @property
+    @override
+    def class_to_idx(self) -> Dict[str, int]:
+        return {label: index for index, label in enumerate(self.classes)}
+
+    @override
+    def filename(self, index: int) -> str:
+        return self._samples_names[index]
+
+    @override
+    def prepare_data(self) -> None:
+        _validators.check_dataset_exists(self._root, True)
+
+    @override
+    def configure(self) -> None:
```

Comment:
I think the `configure` method is not meant to be overwritten, only `setup`, `prepare_data` and `validate`

Reply:
ups, nevermind, I was mistaken, `setup` seems to be the one we shouldn't overwrite